### PR TITLE
Prepare `aster-time` for platform-dependent implementations

### PR DESCRIPTION
**Pull request by YanWQ-monad**
Prepare aster-time for platform-dependent implementations, suggested by https://github.com/asterinas/asterinas/pull/1093#discussion_r1692621996.

Add `aster_time::drivers::Rtc` enum for platform-dependent implementations. An implementation (or, a driver)
1. can use `#[cfg(...)]` to control whether it's compiled on compile-time, for example, CMOS (seems) only exists on x86, so it is compiled only on x86.
2. and can determinate whether to activate on runtime, for example, on RISC-V or some embedding systems, there are different RTC devices on one architecture, so sometimes we need to choose one on runtime. This is done by iterating all compiled drivers and finding the first driver that can initialize successfully.

### CODE REVIEW

File: kernel/comps/time/src/lib.rs [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
 
 mod clocksource;
-mod rtc;
+mod drivers;
 mod tsc;
 
 pub const NANOS_PER_SECOND: u32 = 1_000_000_000;
 pub static VDSO_DATA_HIGH_RES_UPDATE_FN: Once<Arc<dyn Fn(Instant, u64) + Sync + Send>> =
     Once::new();
+static RTC_DRIVER: Once<Rtc> = Once::new();
```
Comment (cchanging) [inline kernel/comps/time/src/lib.rs:27]: Can we declare `RTC_DRIVER` as `Once<Arc<dyn coms::Driver + Send + Sync>>`, thus we do not have to use a macro to implement a enum type `RTC`. In addition, the initialization actions can put in a `init` function in the drivers module.

File: kernel/comps/time/src/drivers/mod.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+use crate::SystemTime;
+
+/// Generic interface for RTC drivers
+pub trait Driver: Sized {
+    /// Create a RTC driver.
+    /// Returns [`Some<Self>`] on success, [`None`] otherwise (e.g. platform unsupported).
+    fn try_new() -> Option<Self>;
+
+    /// Read RTC
```
Comment (cchanging) [inline kernel/comps/time/src/drivers/mod.rs:11]: Add the full stop. Check the other documents.

File: kernel/comps/time/src/drivers/cmos.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            year,
+            month,
+            day,
+            hour,
+            minute,
+            second,
+        }
+    }
+
+    /// convert BCD to binary values
```
Comment (cchanging) [inline kernel/comps/time/src/drivers/cmos.rs:60]: -> `Converts BCD to binary values.`
Documents should begin with the third person singular and should begin with a capital letter. These are actually issues that existed before, but the code changes in this PR can be corrected if it's convenient.

**Review by cchanging — COMMENTED** [review]
Good job! I have only several suggestions.

**Review by cchanging — APPROVED** [review]
LGTM!

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**